recipes/cluster.rb

```diff
@@ -0,0 +1,44 @@
+#
+# Cookbook Name:: rabbitmq
+# Recipe:: cluster
+#
+# Author: Sunggun Yu <sunggun.dev@gmail.com>
+# Copyright (C) 2015 Sunggun Yu
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'json'
+
+include_recipe 'rabbitmq::default'
+
+cluster_nodes = node['rabbitmq']['clustering']['cluster_nodes']
+cluster_nodes = cluster_nodes.to_json
+
+if node['rabbitmq']['cluster']
+  # Manual clustering
+  unless node['rabbitmq']['clustering']['use_auto_clustering']
+    # Join in cluster
+    rabbitmq_cluster cluster_nodes do
+      action :join
+    end
```

Comment:
@jesseadams 
node['rabbitmq']['clustering']['use_auto_clustering'] is the flag for using auto clustering or manual clustering.  should I use `not_if`? can you please explain to me how `not_if` different from `unless` at this code?